bin/slack/main.rb

```diff
@@ -0,0 +1,128 @@
+#             "title": "The Further Adventures of Slackbot",
+#             "fields": [
+#                 {
+#                     "title": "Volume",
+#                     "value": "1",
+#                     "short": true
+#                 },
+#                 {
+#                     "title": "Issue",
+#                     "value": "3",
+#             "short": true
+#                 }
+#             ],
+#             "author_name": "Stanford S. Strickland",
+#             "author_icon": "http://a.slack-edge.com/7f18https://a.slack-edge.com/a8304/img/api/homepage_custom_integrations-2x.png",
+#             "image_url": "http://i.imgur.com/OJkaVOI.jpg?1"
+#         },
+#         {
+#             "title": "Synopsis",
+#             "text": "After @episod pushed exciting changes to a devious new branch back in Issue 1, Slackbot notifies @don about an unexpected deploy..."
+#         },
+#         {
+#             "fallback": "Would you recommend it to customers?",
+#             "title": "Would you recommend it to customers?",
+#             "callback_id": "comic_1234_xyz",
+#             "color": "#3AA3E3",
+#             "attachment_type": "default",
+#             "actions": [
+#                 {
+#                     "name": "recommend",
+#                     "text": "Recommend",
+#                     "type": "button",
+#                     "value": "recommend"
+#                 },
+#                 {
+#                     "name": "no",
+#                     "text": "No",
+#                     "type": "button",
+#                     "value": "bad"
+#                 }
+#             ]
+#         }
+#     ]
+# }
+
+conn = CI::Connector.from_env
+conn.on('environment.lifecycle') do |event|
+  if event['event'] == 'release_completed'
+
+    uri = URI("https://hooks.slack.com/services/T03AA3Y27/BFP2H3YV9/" + ENV['SLACK_TOKEN'])
+
+    header = {
+      'Content-Type': 'application/json',
+    }
+
+    payload = {
+      text: "Release #{event['data']['project']['name']} #{event['data']['name']}",
+      "attachments": [
+        {
+          "color": "#36a64f",
+          "title": "Release completed",
+          "ts": DateTime.iso8601(date).to_time.to_i,
+          "fields": [
+            {
+              "title": "Project",
+              "value": event['data']['project']['name'],
+              "short": flase
+            },
+            {
+              "title": "Release",
+              "value": event['data']['name'],
+              "short": false
+            },
+            {
+              "title": "Commit",
+              "value": event['data']['commitId'],
+              "short": false
+            }
+          ]
+        }
+      ]
+    }
+
+    # Create the HTTP objects
+    http = Net::HTTP.new(uri.host, uri.port)
+    http.use_ssl = true
+    request = Net::HTTP::Post.new(uri.request_uri, header)
+    puts payload.to_json
+    request.body = payload.to_json
+
+    # Send the request
+    http.request(request)
```

Comment:
Why we don't check the response from slack here?

Reply:
no, I don't have kafka ack either

Reply:
@tauffredou How we get error notification when this post on slack failed?